demo/Google.Storage.V1.Demo/Program.cs

```diff
@@ -13,175 +13,180 @@
-            var results = await client.ListAllObjectsAsync(bucket, prefix, new ListObjectsOptions { PageSize = 3 });
-            foreach (var obj in results)
+            if (parentFolder == "") { Console.WriteLine($"Files and folders in bucket {bucket}:"); }
+
+            string shortName = parentFolder == "" ? bucket : BucketHelper.ShortName(parentFolder);
+            Console.WriteLine($"{indent}{shortName}/");
+            indent += "  ";
+
+            foreach (var file in client.ListFiles(bucket, parentFolder))
             {
-                Console.WriteLine($"  {obj.Name}");
+                Console.WriteLine($"{indent}{file.ShortName()} [{file.ContentType}]");
+            }
+
+            foreach (var folder in client.ListFolders(bucket, parentFolder))
+            {
+                ListFilesAndFolders(client, bucket, folder, indent);
             }
         }
 
-        private static async Task DownloadObject(StorageClient client, string bucket, string source, string destination)
+        static void DownloadFile(StorageClient client, string bucket, string source, string destination)
         {
-            using (var output = File.Create(destination))
+            Console.WriteLine($"Downloading GCS file {source} from bucket {bucket} to local file {destination}:");
+            using (var stream = File.Create(destination))
             {
                 var progress = new Progress<IDownloadProgress>(
-                    p => Console.WriteLine($"Downloaded {p.BytesDownloaded} bytes; status: {p.Status}"));
+                    p => Console.WriteLine($"  Downloaded {p.BytesDownloaded} bytes; status: {p.Status}")
+                );
 
-                await client.DownloadObjectAsync(bucket, source, output,
-                    new DownloadObjectOptions { ChunkSize = 256 * 1024 },
-                    CancellationToken.None, progress);
+                client.DownloadObject(bucket, source, stream, null, progress);
             }
         }
 
-        private static async Task UploadObject(StorageClient client, string source, string bucket, string destination, string contentType)
+        static StorageObject UploadFile(StorageClient client, string bucket, string source, string destination, string contentType, bool makePublic)
         {
-            using (var input = File.OpenRead(source))
+            Console.WriteLine($"Uploading local file {source} of type {contentType} to GCS file {destination} in {bucket}:");
+            using (var stream = File.OpenRead(source))
             {
                 var progress = new Progress<IUploadProgress>(
-                    p => Console.WriteLine($"Uploaded {p.BytesSent} bytes; status: {p.Status}"));
+                  p => Console.WriteLine($"  Uploaded {p.BytesSent} bytes; status: {p.Status}")
+                );
+
+                var options = new UploadObjectOptions
+                {
+                    PredefinedAcl = makePublic ?
```

Comment:
Out of interest, how hard was it for you to find the right values to use here? This always felt a bit ugly to me, but introducing another enum didn't appeal much either.